Collection.js

```diff
@@ -0,0 +1,122 @@
+/*!
+ * Copyright (c) 2020 Digital Bazaar, Inc. All rights reserved.
+ */
+'use strict';
+
+import {EdvClient, EdvDocument} from 'edv-client';
+
+const JWE_ALG = 'ECDH-ES+A256KW';
+
+export default class Collection {
+  constructor({type, capability, client, invocationSigner}) {
+    this.type = type;
+    this.capability = capability;
+    this.client = client;
+    this.invocationSigner = invocationSigner;
+  }
+
+  async create({item, meta} = {}) {
+    if(!(item && typeof item === 'object')) {
+      throw new TypeError(`"item" must be an object.`);
+    }
+    if(!(item.type === this.type ||
+      (Array.isArray(item.type) && item.type.includes(this.type)))) {
+      throw new TypeError(
+        `"item.type" (${item.type}) must include "${this.type}".`);
+    }
+    const edvDoc = await this._getEdvDocument();
+    const id = await EdvClient.generateId();
+    return await edvDoc.write({doc: {id, content: item, meta}});
+  }
+
+  async get({id} = {}) {
+    if(typeof id !== 'string') {
+      throw new TypeError('"id" or must be given.');
+    }
+    const results = await this._findDocuments({id});
+    if(results.length > 0) {
+      return results[0];
+    }
+    return null;
+  }
+
+  async getAll() {
+    const {instance, capability} = this;
+    const results = await this._findDocuments(
+      {instance, type: this.type, capability});
+    return results;
+  }
+
+  async update({item, meta} = {}) {
+    if(!(item && typeof item === 'object')) {
+      throw new TypeError(`"item" must be an object.`);
+    }
+    if(!(item.type === this.type ||
+      (Array.isArray(item.type) && item.type.includes(this.type)))) {
+      throw new TypeError(
+        `"item.type" (${item.type}) must include "${this.type}".`);
+    }
+    const existing = await this.get({id: item.id});
+    const edvDoc = await this._getEdvDocument({id: existing.id});
+    const updatedDoc = {
+      ...existing
+    };
+    updatedDoc.content = item;
+    if(meta) {
+      updatedDoc.meta = meta;
+    }
+    return await edvDoc.write({
+      doc: updatedDoc
+    });
+  }
+
+  async remove({id} = {}) {
+    if(typeof id !== 'string') {
+      throw new TypeError('"id" or must be given.');
```

Comment:
```suggestion
      throw new TypeError('"id" must be a document ID.');
```

Reply:
Updated; it's also not a doc ID -- it's an ID for the item or content. I went with the general pattern of just saying it needs to be a string.